pkg/collector/corechecks/system/cpu/cpu/cpu.go

```diff
@@ -40,40 +39,81 @@ func (c *Check) Run() error {
 	if err != nil {
 		return err
 	}
+	c.reportContextSwitches(sender)
+	numCores, err := c.reportCpuInfo(sender)
+	if err != nil {
+		return err
+	}
+	err = c.reportCpuMetricsPercent(sender, numCores)
+	if err != nil {
+		return err
+	}
+	err = c.reportCpuMetricsPerCpu(sender)
+	if err != nil {
+		return err
+	}
+	sender.Commit()
+	return nil
+}
 
-	err = collectCtxSwitches(sender)
+func (c *Check) reportContextSwitches(sender sender.Sender) {
+	ctxSwitches, err := getContextSwitches()
 	if err != nil {
-		log.Debugf("cpu.Check could not read context switches: %s", err.Error())
-		// Don't return here, we still want to collect the CPU metrics even if we could not
+		log.Debugf("could not read context switches: %s", err.Error())
+		// Don't return error here, we still want to collect the CPU metrics even if we could not
 		// read the context switches
+	} else {
+		log.Debugf("getContextSwitches: %d", ctxSwitches)
+		sender.MonotonicCount("system.cpu.context_switches", float64(ctxSwitches), "", nil)
 	}
+}
 
-	cpuTimes, err := cpuTimesFunc(false)
+func (c *Check) reportCpuInfo(sender sender.Sender) (numCores int32, err error) {
+	cpuInfo, err := getCpuInfo()
 	if err != nil {
-		log.Errorf("cpu.Check: could not retrieve cpu stats: %s", err)
+		log.Errorf("could not retrieve cpu stats: %s", err.Error())
```

Comment:
nit: it could be better to make these log lines to be more specific, for instance `could not retrieve cpu info` instead of `could not retrieve cpu stats`